app/services/container_dashboard_service.rb

```diff
@@ -160,4 +161,43 @@ def heatmaps
       }
     }
   end
+
+  def node_utilization
+    resource_ids = @ems.present? ? [@ems.id] : ManageIQ::Providers::ContainerManager.all.pluck(:id)
+    metrics = VimPerformanceDaily.find_entries({:tz => @controller.current_user.get_timezone})
+    metrics = metrics.where(:resource_type => 'ExtManagementSystem', :resource_id => resource_ids)
+    metrics = metrics.order("timestamp")
+
+    used_cpu = Hash.new { |hash, key| hash[key] = 0 }
+    used_mem = used_cpu.dup
+    total_cpu = used_cpu.dup
+    total_mem = used_cpu.dup
```

Comment:
This doesn't seem readable (easily understandable) at first sight. If you want to spare the repetition of `Hash.new { |hash, key| hash[key] = 0 }` just make it a function.